src/test/regress/sql/seclabel.sql

```diff
@@ -0,0 +1,105 @@
+--
+-- SECLABEL
+--
+-- Test suite for SECURITY LABEL ON ROLE statements
+--
+
+-- first we remove one of the worker nodes to be able to test
+-- citus_add_node later
+SELECT citus_remove_node('localhost', :worker_2_port);
+
+-- now we register a label provider
+CREATE FUNCTION citus_test_register_label_provider()
+  RETURNS void
+  LANGUAGE C
+  AS 'citus', $$citus_test_register_label_provider$$;
+SELECT citus_test_register_label_provider();
+
+CREATE ROLE user1;
+
+-- check an invalid label for our current dummy hook citus_test_object_relabel
+SECURITY LABEL ON ROLE user1 IS 'invalid_label';
+
+-- if we disable metadata_sync, the command will not be propagated
+SET citus.enable_metadata_sync TO off;
+SECURITY LABEL ON ROLE user1 IS 'citus_unclassified';
+SELECT objtype, objname, provider, label FROM pg_seclabels;
+
+\c - - - :worker_1_port
+SELECT objtype, objname, provider, label FROM pg_seclabels;
+
+\c - - - :master_port
+SELECT citus_test_register_label_provider();
+
+-- check that we only support propagating for roles
+SET citus.shard_replication_factor to 1;
+CREATE TABLE a (a int);
+SELECT create_distributed_table('a', 'a');
+SECURITY LABEL ON TABLE a IS 'citus_classified';
+DROP TABLE a;
+
+-- the registered label provider is per session only
+-- this means that we need to maintain the same connection to the worker node
+-- in order for the label provider to be visible there
+-- hence here we create the necessary session_level_connection_to_node functions
+SET citus.enable_metadata_sync TO off;
+CREATE OR REPLACE FUNCTION start_session_level_connection_to_node(text, integer)
+        RETURNS void
+        LANGUAGE C STRICT VOLATILE
+        AS 'citus', $$start_session_level_connection_to_node$$;
+CREATE OR REPLACE FUNCTION override_backend_data_gpid(bigint)
+        RETURNS void
+        LANGUAGE C STRICT IMMUTABLE
+        AS 'citus', $$override_backend_data_gpid$$;
+SELECT run_command_on_workers($$SET citus.enable_metadata_sync TO off;CREATE OR REPLACE FUNCTION override_backend_data_gpid(bigint)
+        RETURNS void
+        LANGUAGE C STRICT IMMUTABLE
+        AS 'citus'$$);
+CREATE OR REPLACE FUNCTION run_commands_on_session_level_connection_to_node(text)
+        RETURNS void
+        LANGUAGE C STRICT VOLATILE
+        AS 'citus', $$run_commands_on_session_level_connection_to_node$$;
+CREATE OR REPLACE FUNCTION stop_session_level_connection_to_node()
+        RETURNS void
+        LANGUAGE C STRICT VOLATILE
+        AS 'citus', $$stop_session_level_connection_to_node$$;
```

Comment:
Should we do those in multi_test_helpers.sql or such?

Reply:
I went with the GUC option for now, if we decide to get back on this, I will see if I can include this in test helpers.